test/special-browsers.test.ts

```diff
@@ -0,0 +1,107 @@
+import connect from 'connect';
+import request from 'supertest';
+import { IncomingMessage, ServerResponse } from 'http';
+
+import parseCsp from 'content-security-policy-parser';
+
+import AGENTS from './browser-data.json';
+
+import csp = require('..');
+
+function makeApp (middleware: ReturnType<typeof csp>) {
+  const result = connect();
+  result.use(middleware);
+  result.use((_req: IncomingMessage, res: ServerResponse) => {
+    res.end('Hello world!');
+  });
+  return result;
+}
+
+describe('special browsers', () => {
+  [
+    'Firefox 22',
+    'Firefox OS 1.4',
+    'Firefox for Android 24',
+  ].forEach((browser) => {
+    describe(browser, () => {
+      it('sets the header properly', (done) => {
+        const app = makeApp(csp({
+          directives: {
+            defaultSrc: ["'self'"],
+            connectSrc: ['connect.com'],
+          },
+        }));
+
+        request(app).get('/').set('User-Agent', AGENTS['Firefox 22'].string)
```

Comment:
Whoops—this shouldn't always be the Firefox 22 user-agent! This was pre-existing, but would you mind changing this to `AGENTS[browser as keyof typeof AGENTS].string`?